DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -1,244 +1,297 @@
+    kD0Size = 13,                                                             // Width of D0
+    kTrackParamsSize = kD0Size + kZ0Size + kTanlSize + kPhiSize + kRinvSize,  // Width of track parameters
+
+    kChi2RPhiSize = 4,    // Width of Chi2 for r-phi
+    kChi2RZSize = 4,      // Width of Chi2 for r-z
+    kBendChi2Size = 3,    // Width of the Bend-Chi2
+    kHitPatternSize = 7,  // Width of the hit pattern for stubs
+    kMVAQualitySize = 3,  // Width of track quality MVA
+    kMVAOtherSize = 6,    // Space for two specialized MVA selections
+    kTrackQualitySize = kMVAOtherSize + kMVAQualitySize + kHitPatternSize + kBendChi2Size + kChi2RZSize +
+                        kChi2RPhiSize,  // Width of track quality
+
+    kTrackWordSize = kTrackQualitySize + kTrackParamsSize + kValidSize,  // Width of the Track Finding Word in bits
+  };
+
+  enum TrackBitLocations {
+    // The location of the least significant bit (LSB) and most significant bit (MSB) in the track word for different fields
+    kValidLSB = 0,
+    kValidMSB = kValidLSB + TrackBitWidths::kValidSize - 1,
+    kRinvLSB = kValidMSB + 1,
+    kRinvMSB = kRinvLSB + TrackBitWidths::kRinvSize - 1,
+    kPhiLSB = kRinvMSB + 1,
+    kPhiMSB = kPhiLSB + TrackBitWidths::kPhiSize - 1,
+    kTanlLSB = kPhiMSB + 1,
+    kTanlMSB = kTanlLSB + TrackBitWidths::kTanlSize - 1,
+    kZ0LSB = kTanlMSB + 1,
+    kZ0MSB = kZ0LSB + TrackBitWidths::kZ0Size - 1,
+    kD0LSB = kZ0MSB + 1,
+    kD0MSB = kD0LSB + TrackBitWidths::kD0Size - 1,
+    kChi2RPhiLSB = kD0MSB + 1,
+    kChi2RPhiMSB = kChi2RPhiLSB + TrackBitWidths::kChi2RPhiSize - 1,
+    kChi2RZLSB = kChi2RPhiMSB + 1,
+    kChi2RZMSB = kChi2RZLSB + TrackBitWidths::kChi2RZSize - 1,
+    kBendChi2LSB = kChi2RZMSB + 1,
+    kBendChi2MSB = kBendChi2LSB + TrackBitWidths::kBendChi2Size - 1,
+    kHitPatternLSB = kBendChi2MSB + 1,
+    kHitPatternMSB = kHitPatternLSB + TrackBitWidths::kHitPatternSize - 1,
+    kMVAQualityLSB = kHitPatternMSB + 1,
+    kMVAQualityMSB = kMVAQualityLSB + TrackBitWidths::kMVAQualitySize - 1,
+    kMVAOtherLSB = kMVAQualityMSB + 1,
+    kMVAOtherMSB = kMVAOtherLSB + TrackBitWidths::kMVAOtherSize - 1,
+  };
+
+  // Binning constants
+  static constexpr double minRinv = -0.00852662681;
+  static constexpr double minPhi0 = -0.69813248;  // relative to the center of the sector
+  static constexpr double minTanl = -8.;
+  static constexpr double minZ0 = -20.46912512;
+  static constexpr double minD0 = -16.;
+
+  static constexpr double stepRinv = (2. * std::abs(minRinv)) / (1 << TrackBitWidths::kRinvSize);
+  static constexpr double stepPhi0 = (2. * std::abs(minPhi0)) / (1 << TrackBitWidths::kPhiSize);
+  static constexpr double stepTanL = (1. / (1 << 12));
+  static constexpr double stepZ0 = (2. * std::abs(minZ0)) / (1 << TrackBitWidths::kZ0Size);
+  static constexpr double stepD0 = (1. / (1 << 8));
+
+  const std::vector<double> chi2RPhiBins = {
```

Comment:
since these `vector`s are constants, would it make sense to replace them with `std::array`s ?

Reply:
These vectors are used as a parameter to:
```c++
unsigned int getBin(double value, const std::vector<double>& bins) const
```
Unless we also want to template this function (to pass the size as a template parameter) I wouldn't recommend changing the types.